ros1_foxglove_bridge/src/ros1_foxglove_bridge_nodelet.cpp

```diff
@@ -665,27 +667,49 @@ class FoxgloveBridge : public nodelet::Nodelet {
           success = false;
         }
       }
+      if (_invalidParams.find(paramName) != _invalidParams.end()) {
+        continue;
+      }
 
       try {
         XmlRpc::XmlRpcValue value;
         getMTNodeHandle().getParam(paramName, value);
         params.push_back(fromRosParam(paramName, value));
       } catch (const std::exception& ex) {
         ROS_ERROR("Invalid parameter '%s': %s", paramName.c_str(), ex.what());
+        invalidParams.push_back(paramName);
         success = false;
       } catch (const XmlRpc::XmlRpcException& ex) {
         ROS_ERROR("Invalid parameter '%s': %s", paramName.c_str(), ex.getMessage().c_str());
+        invalidParams.push_back(paramName);
         success = false;
       } catch (...) {
         ROS_ERROR("Invalid parameter '%s'", paramName.c_str());
+        invalidParams.push_back(paramName);
         success = false;
       }
     }
 
     _server->publishParameterValues(hdl, params, requestId);
 
     if (!success) {
-      throw std::runtime_error("Failed to retrieve one or multiple parameters");
+      for (std::string& param : invalidParams) {
+        _invalidParams.insert(param);
+      }
+
+      if (!invalidParams.empty()) {
+        std::ostringstream errorMsg;
+        errorMsg << "Failed to retrieve the following parameters: ";
```

Comment:
for something simple like this, I imagine you could use `+=` on a std::string rather than a stringstream, however this is fine too